src/Nervos/Transaction.cpp

```diff
@@ -0,0 +1,362 @@
+        uint64_t availableAmount =
+            std::accumulate(signingInputCells.begin(), signingInputCells.end(), uint64_t(0),
+                            [&](const uint64_t total, const Cell& cell) {
+                                if (cell.type.empty()) {
+                                    inputCells.emplace_back(std::move(cell));
+                                    return total + cell.capacity;
+                                } else {
+                                    return total;
+                                }
+                            });
+        uint64_t fee =
+            (sizeWithoutInputs() + inputCells.size() * sizeOfSingleInputAndWitness()) * byteFee;
+        outputs[outputs.size() - 1].capacity += availableAmount - fee;
+    } else {
+        // Transfer exact number of tokens
+        auto sorted = signingInputCells;
+        std::sort(sorted.begin(), sorted.end(),
+                  [](const Cell& lhs, const Cell& rhs) { return lhs.capacity < rhs.capacity; });
+        uint64_t fee =
+            (sizeWithoutInputs() + inputCells.size() * sizeOfSingleInputAndWitness()) * byteFee;
+        uint64_t feeForChangeOutput = sizeOfSingleOutput(changeAddress);
+        uint64_t sum = std::accumulate(
+            inputCells.begin(), inputCells.end(), uint64_t(0),
+            [&](const uint64_t total, const Cell& cell) { return total + cell.capacity; });
+        bool gotEnough = false;
+        for (auto& cell : sorted) {
+            if (!cell.type.empty()) {
+                continue;
+            }
+            inputCells.emplace_back(std::move(cell));
+            sum += cell.capacity;
+            fee += sizeOfSingleInputAndWitness() * byteFee;
+            if (sum >= requiredAmount + fee) {
+                gotEnough = true;
+                uint64_t remainingAmount = sum - requiredAmount - fee;
+                if (remainingAmount >= feeForChangeOutput + minCellCapacityForNativeToken) {
+                    // If change is enough, add it to the change address
+                    outputs.emplace_back(remainingAmount - feeForChangeOutput,
+                                         Script(changeAddress), Script());
+                    outputsData.emplace_back();
+                } else {
+                    // If change is not enough, add it to the destination address
+                    outputs[outputs.size() - 1].capacity += remainingAmount;
+                }
+                break;
+            }
+        }
+        if (!gotEnough) {
+            return Common::Proto::Error_not_enough_utxos;
+        }
+    }
+    return Common::Proto::OK;
+}
+
+Common::Proto::SigningError Transaction::planSudtTransfer(uint256_t amount,
+                                                          const Cells& signingInputCells,
+                                                          const Address& changeAddress,
+                                                          uint64_t byteFee) {
```

Comment:
same remark as above, the code is almost unreadeable

Reply:
Done